### DISCUSSION TITLE
Allow editing the condition for a breakpoint if it exceeds 200 hits

### DISCUSSION THREAD

**Discussion by jaril** [RecordReplay/devtools#3273] [Ideas]
Right now, if a breakpoint has over 200 hits, we restrict editing the breakpoint. This includes the log message and the condition.

There are cases where the user bumps into our 200 hit limitation, and knows how to be more specific about which hits they're interested in. When that happens, they're stuck. They can't edit the condition.

[This Replay](https://app.replay.io/?id=c859afd3-1040-4fdb-a9b2-a04bdc76c21c) has some good lines on script.js to add a breakpoint on to test this. 

## Examples

**A1: Hovering on a line with <200 hits (the button is clickable)**
![image](https://user-images.githubusercontent.com/15959269/126408060-87e57d53-4026-4412-9376-e426c674d145.png)

**A2: After clicking the button on a line with <200 hits**
![image](https://user-images.githubusercontent.com/15959269/126408154-f6e45169-6320-40da-92e1-2b18813ab038.png)

**B1: Hovering on a line with >200 hits (the button is disabled)**
![image](https://user-images.githubusercontent.com/15959269/126408089-62bb26a5-bcb0-4c61-8285-8f5bfe186a93.png)

**Comment by jasonLaster** [comment]
The challenge here is that editing the condition is just as expensive computationally as editing the log. 

I think we have two options at the moment:

1. Add an advanced setting that lets users edit this (perhaps in the paid plan)
2. Add a "trim" features so that users can focus in on a region

**Comment by jonbell-lot23** [comment]
Agreed, @jasonLaster . I spoke with @jaril earlier and I think we can put this lower in our priority stack for now. Still important, but not as important the other projects we've got going on.

**Comment by bhackett1024** [comment]
It would be nice to let users see information about at least some of the hits when there are a lot of them, e.g. have a dialog "This breakpoint has 3680 hits but we can only show 200 at a time, select the range of hits you're interested in" or something.  I agree it's not a high priority for now.

Eventually we'll be able to handle conditions more efficiently in the backend and be able to use them with breakpoints that are hit millions of times, but that's also not a priority right now.  Regardless, the user should have options for dealing with breakpoints that are hit many times without having to set a condition.

**Comment by jasonLaster** [comment]
> It would be nice to let users see information about at least some of the hits when there are a lot of them, e.g. have a dialog "This breakpoint has 3680 hits but we can only show 200 at a time, select the range of hits you're interested in" or something. I agree it's not a high priority for now.

Agreed. We now that we have API support for running an analysis on an arbitrary set of points we should be able to do that

**Comment by jonbell-lot23** [comment]
March 22, 2022 update:

We now have Focus Mode! That can be a big part of the messaging here. I have an item to address this.
